src/controllers/good-type.controller.ts

```diff
@@ -0,0 +1,415 @@
+import {service} from '@loopback/core';
+import {Filter, repository} from '@loopback/repository';
+import {
+  get,
+  getModelSchemaRef,
+  HttpErrors,
+  param,
+  patch,
+  post,
+  put,
+  requestBody,
+} from '@loopback/rest';
+import {GoodType} from '../models';
+import {GoodTypeRepository} from '../repositories';
+import {ProofOfGoodSmartContractService} from '../services';
+
+export class GoodTypeController {
+  constructor(
+    @repository(GoodTypeRepository)
+    public goodTypeRepository: GoodTypeRepository,
+    @service(ProofOfGoodSmartContractService)
+    private proofOfGoodSmartContractService: ProofOfGoodSmartContractService,
+  ) {}
+
+  @post('/good-types', {
```

Comment:
we took the convention to have the route remove the "good-" part
```suggestion
  @post('/types', {
```